### PR TITLE
x/incentives: test improvements

### PR DESCRIPTION
**Pull request by czarcas7ic**
Part of: #1837 

## What is the purpose of the change

Audit, clean up, and test modifications for the incentives module

This is part 3 of 4:

1. Proto changes
2. CLI fixes
3. Tests
4. All other changes

## Brief Changelog
 
- Remove root folder README in favor of spec folder
- Fixes reward estimation query to work as intended (pulls owners lockIDs if not provided)
- Allow benchmark to work again
- Adds TestSyntheticDistribute

## Testing and Verifying

This change is compromised of test modifications

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/incentives/README.md [hunk 1]
```diff
@@ -1,84 +0,0 @@
-# Incentives Module
```
Comment (mattverse) [inline x/incentives/README.md:1]: Assuming that this is to be replaced and updated via `spec/README.md`

Let's make sure we revisit some of these contents as these were for easier integration for external incentivizers.

**Review by mattverse — APPROVED** [review]
Overall LGTM! Nice work on all the detailed comments and documentations on the comments as well!

I'd prefer removing the TODOs from the code and create / manage a separate issue for it instead of merging with the different TODOs!

File: x/incentives/keeper/grpc_query_test.go [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
 
+// TestGRPCGaugeByID tests querying gauges via gRPC returns the correct response.
 func (suite *KeeperTestSuite) TestGRPCGaugeByID() {
 	suite.SetupTest()
 
 	// create a gauge
 	gaugeID, _, coins, startTime := suite.SetupNewGauge(false, sdk.Coins{sdk.NewInt64Coin("stake", 10)})
 
-	// Ensure that a querying for a gauge with an ID that doesn't exist returns an error
+	// ensure that a querying for a gauge with an ID that doesn't exist returns an error
```
Comment (p0mvn) [inline x/incentives/keeper/grpc_query_test.go:23]:
```suggestion
	// ensure that querying for a gauge with an ID that doesn't exist returns an error.
```

File: x/incentives/keeper/store_test.go [hunk 3]
```diff
@@ -1,24 +1,35 @@
...
+	// remove gauge 1 from key2, resulting in a reduction from 3 to 2 entries
 	err := suite.App.IncentivesKeeper.DeleteGaugeRefByKey(suite.Ctx, key2, 1)
 	suite.Require().NoError(err)
 
+	// ensure key2 now only has 2 entires
 	gaugeRefs3 := suite.App.IncentivesKeeper.GetGaugeRefs(suite.Ctx, key2)
 	suite.Require().Equal(len(gaugeRefs3), 2)
 }
+
+var _ = suite.TestingSuite(nil)
```
Comment (p0mvn) [inline x/incentives/keeper/store_test.go:35]: let's move up please

**Review by p0mvn — APPROVED** [review]

File: x/incentives/README.md (hunk 1 above)
Comment (czarcas7ic) [reply x/incentives/README.md:1]: The specific gauge examples still lives in the spec folder under the "create gauge" tx example

https://github.com/osmosis-labs/osmosis/tree/main/x/incentives/spec#create-gauge-1

Do you think we should link this maybe in the intro so integrators can get to it easier?

File: x/incentives/keeper/grpc_query_test.go [hunk 4]
```diff
@@ -320,39 +348,50 @@ func (suite *KeeperTestSuite) TestGRPCRewardsEst() {
...
 		EndEpoch: 10,
 	})
 	suite.Require().NoError(err)
 	suite.Require().Equal(res.Coins, coins)
 
+	// after the current epoch ends, mint more coins that matches the lock coin demon created earlier
 	epochIdentifier := suite.App.MintKeeper.GetParams(suite.Ctx).EpochIdentifier
 	curEpochNumber := suite.App.EpochsKeeper.GetEpochInfo(suite.Ctx, epochIdentifier).CurrentEpoch
 	suite.App.EpochsKeeper.AfterEpochEnd(suite.Ctx, epochIdentifier, curEpochNumber)
 	// TODO: Figure out what this number should be
```
[the remark this answers is not in the document]
Comment (czarcas7ic) [reply x/incentives/keeper/grpc_query_test.go:389]: I have no idea, @ValarDragon do you know what you meant when you wrote this (was ~13 months ago)? It seems the 1500000 int works fine, unsure why it should/would change.

File: x/incentives/keeper/distribute_test.go [hunk 5]
```diff
@@ -36,82 +35,177 @@ func (suite *KeeperTestSuite) TestDistribute() {
...
 		}
 	}
 
-	// TODO: test distribution for synthetic lockup as well
 }
 
-// TODO: Make this test table driven, or move whatever it tests into
-// the much simpler TestDistribute
+// TODO: Make this test table driven, or move whatever it tests into the much simpler TestDistribute
+// TODO: Make issue for the above
```
[the remark this answers is not in the document]
Comment (czarcas7ic) [reply x/incentives/keeper/distribute_test.go:179]: https://github.com/osmosis-labs/osmosis/issues/1995

Done!

**Comment by p0mvn** [comment]
Can we merge this @czarcas7ic ?

File: x/incentives/keeper/grpc_query_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (ValarDragon) [reply x/incentives/keeper/grpc_query_test.go:391]: I don't recall

**Pull request merged**